coursier.bzl

```diff
@@ -776,6 +776,9 @@ def _coursier_fetch_impl(repository_ctx):
                     # So, we use rfind to get the index of the final '@' in the
                     # repository url instead.
                     primary_artifact_path = primary_url[len(url):]
+                elif url.find("@") == -1 and primary_url.find("@") != -1:
+                    username_len = primary_url.find("//") + 2 + primary_url.find("@") + 1
```

Comment:
Could you add some comments here to help visualize the values? And also a comment on when this codepath will be used.